_drafts/intro-to-working-with-data-in-CUBA.adoc

```diff
@@ -0,0 +1,208 @@
+
+In this case, the attribute `identificationNumber` is not the primary key of the entity. But it represents a globally unique identification number given by proper Pet authorities - therefore it is modelled uniquely within the application. 
+
+If the task is to load a Pet by its primary key (`id`) the query is easier to create as the query part is not needed: `dataManager.load(Pet.class).id(id).one();`
+
+=== Multiple Items by Query
+
+The next very common option is to load multiple items defined by a query.
+
+The Petclinic application has the capability to let the user send out Disease Warning Mailings. The user has to define the location of a known disease as well as the type of Pet that is endangered by this disease. The system will find all endangered Pets and send out a Mailing to the owners.
+
+image:https://dummyimage.com/600x400/00b5ec/ffffff.png&text=Disease Warning Mailing[] Disease Warning Mailing overview
+
+
+The part that finds the endangered Pets looks like this with the use of the DataManager API:
+
+.DiseaseWarningMailingServiceBean.java
+[source,java]
+----
+@Inject
+DataManager dataManager;
+
+private List<Pet> findEndangeredPets(PetType petType, String city) {
+    return dataManager.load(Pet.class)
+                .query("select e from petclinic$Pet e where e.owner.city = :ownerCity and e.type.id = :petType")
+                .parameter("ownerCity", city)
+                .parameter("petType", petType)
+                .view("pet-with-owner-and-type")
+                .list();
+}
+----
+
+The query part is a little more complicated compared to above, but the same pattern applies. The differences in this example are the usage of the `list()` method as we expect multiple Pets from the database query. Also a specific view has been used, because not only the Pet data, but also the Owner data has to be loaded in order to send out the Mailings.
+
+== Write Data with DataManager
+
+DataManager is also responsible for writing data into the database. There are also different ways to achieve it based on the case at hand.
+
+=== Single Item
+
+The first most commonly used case is to update or create a single item in the database. There are no major differences between creating and updating of an entity from the usage perspective.
+
+When revisiting the example from above, where a new Visit for a given Pet ID is created, we only looked at the part which loads the Pet by its Pet ID. In this case, we are more interested in the creation part of the Visit entity. The source code to create and save a Visit looks like this:
+
+
+.VisitServiceBean.java
+[source,java]
+----
+@Inject
+DataManager dataManager;
+
+private Visit createVisitForPet(Pet pet) {
+
+    Visit visit = dataManager.create(Visit.class); // <1>
+
+    visit.setPet(pet);
```

Comment:
Thanks @knstvk for the review. You can feel free to just change it directly in the code if you think that it is equally fast or faster then writing the comment. There is actually a new feature in github to use suggestions in code reviews that will streamline the process.

Example of a suggestion: 
```suggestion
    visit.setPet(null);
```